atomicfu-transformer/src/main/kotlin/kotlinx/atomicfu/transformer/AtomicFUTransformer.kt

```diff
@@ -446,10 +446,10 @@ class AtomicFUTransformer(
                 }
                 if (vh) {
                     // VarHandle is needed for all array element accesses and for regular fields with atomic ops
-                    if (f.hasAtomicOps || f.isArray) vhField(protection, f)
+                    if (f.hasAtomicOps || f.isArray) vhField(protection or ACC_SYNTHETIC, f)
                 } else {
                     // FieldUpdater is not needed for arrays (they use AtomicArrays)
-                    if (f.hasAtomicOps && !f.isArray) fuField(protection, f)
+                    if (f.hasAtomicOps && !f.isArray) fuField(protection or ACC_SYNTHETIC, f)
```

Comment:
There's already ACC_SYNTHETIC set in `protection` at lines 425-428 with the only exception of a private field (where it is not needed). https://youtrack.jetbrains.com/issue/KT-40328 should have been caused by something else. It needs better investigation and the corresponding test-case.

Reply:
Actually, the `protection` for package-private fields didn't include `ACC_SYNTHETIC`. 

And this was the case from the issue :point_down: , the inner class `B` caused an accessor in the bytecode for the private field `state`, so the corresponding `FU` field was marked package-private (non-synthetic). 
```
class A {
    private val state = atomic(0)

    inner class B {
        ...
    }
} 
```
Though as there was no actual access to the `state` from the inner class `JVM_IR` didn't not generate an accessor, so the field was private there.

Reply:
The corresponding case on access modifiers is provided in this PR #154